moq-karp/src/broadcast.rs

```diff
@@ -1,116 +1,175 @@
-use crate::{Audio, Catalog, Error, Result, Track, TrackConsumer, TrackProducer, Video};
+use crate::{Audio, Catalog, Error, Result, Track, TrackProducer, Video};
 
 use moq_async::{spawn, Lock};
 use moq_transfork::{Announced, AnnouncedConsumer, AnnouncedMatch, Session};
+use crate::track::TrackConsumer;
 
 use derive_more::Debug;
 
+#[derive(Clone)]
+struct BroadcastListener {
+	pub session: Session,
+	pub catalog: Lock<CatalogProducer>,
```

Comment:
Why do you need this? I think you delete it and move it to `BroadcastProducerState`.